# Mini Project 1 — Analysis Notebook

**Your name:**  Melanie Song
**Dataset:**  restaurants_cleaned.csv (cleaned from 380K_US_Restaurants.csv)
**Date:**  May 13, 2026

This notebook has four sections. Work through them in order. Each section has instructions and a code cell to fill in. Add markdown cells to explain your thinking as you go — that writing is part of the assignment.

When you're done, publish this notebook to your GitHub repository and submit the URL to Canvas.

In [2]:
# Setup — run this cell first
# If any package is missing, it will install automatically
import subprocess, sys

def install(pkg):
    subprocess.check_call([sys.executable, "-m", "pip", "install", pkg, "-q"])

for pkg in ["pandas", "plotly", "kaleido", "nbformat"]:
    try:
        __import__(pkg)
    except ImportError:
        print(f"Installing {pkg}...")
        install(pkg)

import pandas as pd
import plotly.express as px

print("Setup complete.")

Setup complete.


---

## Section 1 — Overview

Before writing any code, fill in this section. A good Overview tells anyone reading your notebook — including a future employer — what the analysis is about before they see a single chart.

**Dataset:** *This is a dataset of 380K Restaurants (Mostly USA-Based) from Kaggle: https://www.kaggle.com/datasets/kwxdata/380k-restaurants-mostly-usa-based/data. This dataset contains approximately 380,000 restaurant records, including information such as restaurant name, category (cuisine type), rating, and address/location. *

**Why this dataset:** *This project connects to my interest in human-centered design by exploring how cultural identity is reflected and surfaced through food data on digital platforms. Understanding patterns of representation and visibility can help inform how platforms categorize, recommend, and present culturally diverse experiences to users.*

**Three analytical questions:**

1. Which Asian cuisines (e.g., Chinese, Japanese, Thai, Vietnamese, etc.) are most and least represented among restaurants in the dataset?
2. How does the representation of Asian cuisines vary across different U.S. states (e.g., which states have higher or lower proportions of Asian cuisine restaurants)?
3. Do average ratings differ across Asian cuisine types, and are certain cuisines consistently rated higher or lower than others?


**What a practitioner would do with these findings:** *Interested parties can access this to understand the distribution of Asian restaurants across the US. Food finders can also use this to locate some restaurants with good ratings. *

---

## Section 2 — Data Profile

Load your dataset and get a basic picture of what's in it. Answer these questions in a markdown cell below your code:

- How many rows and columns does your dataset have?
- What does each column represent?
- Are there any obvious data quality issues (missing values, unexpected types, inconsistent formatting)?
- Which column or columns will your analysis focus on, and why?

In [3]:
# Load your dataset
# Replace 'your_dataset.csv' with your actual filename.
# The file should be in the same folder as this notebook.
# If you're loading from an API result, replace pd.read_csv() with the appropriate call.
#
# Example (app review dataset from class):
# df = pd.read_csv('app_reviews_demo.csv')

df = pd.read_csv('380K_US_Restaurants.csv')  # ← replace with your filename

print(df.shape)
df.head()

(380358, 13)


/var/folders/nz/k06zv2_91mx8l7l3ws72b6680000gn/T/ipykernel_88688/2514341641.py:9: DtypeWarning: Columns (3,9,10,11,12) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('380K_US_Restaurants.csv')  # ← replace with your filename


,Title,Link,Category,Rating,Website,Phone,Address,Images,Categories,Geo_Coordinates,Time_Zone,Latitude,Longitude
0,Dairy Queen Grill & Chill,https://www.google.com/maps/place/Dairy+Queen+...,Fast food restaurant,3.8,http://www.fourteenfoods.com/?y_source=1_ODk5N...,+1256-496-0404,"Dairy Queen Grill & Chill, 3143 US-280, Alexan...",[{'thumbnail': 'https://lh5.googleusercontent....,"['Fast food restaurant', 'Ice cream shop']","{'latitude': 32.9338695, 'longitude': -85.9704...",America/Chicago,32.93387,-85.970419
1,Jake's Restaurant,https://www.google.com/maps/place/Jake%27s+Res...,American restaurant,4.4,http://jakesonbroad.com/,+1256-234-4300,"Jake's Restaurant, 16 Broad St, Alexander City...",[{'thumbnail': 'https://lh5.googleusercontent....,['American restaurant'],"{'latitude': 32.945406, 'longitude': -85.953806}",America/Chicago,32.945406,-85.953806
2,Carib Kitchen,https://www.google.com/maps/place/Carib+Kitche...,Caribbean restaurant,4.9,https://carib-kitchen.webnode.com/,+1256-392-4433,"Carib Kitchen, 68 Broad St, Alexander City, AL...",[{'thumbnail': 'https://lh5.googleusercontent....,['Caribbean restaurant'],"{'latitude': 32.9446173, 'longitude': -85.954932}",America/Chicago,32.944617,-85.954932
3,Cazadores Mexican Restaurant,https://www.google.com/maps/place/Cazadores+Me...,Mexican restaurant,4.5,NaN,+1256-392-3991,"Cazadores Mexican Restaurant, 910 Cherokee Rd,...",[{'thumbnail': 'https://lh5.googleusercontent....,['Mexican restaurant'],"{'latitude': 32.935643, 'longitude': -85.952365}",America/Chicago,32.935643,-85.952365
4,La Posada Mexican Grill,https://www.google.com/maps/place/La+Posada+Me...,Mexican restaurant,4.3,http://www.laposadamexicangrill.net/,+1256-329-3005,"La Posada Mexican Grill, 3714 US-280, Alexande...",[{'thumbnail': 'https://lh5.googleusercontent....,"['Mexican restaurant', 'Latin American restaur...","{'latitude': 32.926932, 'longitude': -85.9649187}",America/Chicago,32.926932,-85.964919


In [4]:
# Check column types and missing values
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 380358 entries, 0 to 380357
Data columns (total 13 columns):
 #   Column           Non-Null Count   Dtype 
---  ------           --------------   ----- 
 0   Title            380358 non-null  object
 1   Link             345209 non-null  object
 2   Category         380302 non-null  object
 3   Rating           337877 non-null  object
 4   Website          285435 non-null  object
 5   Phone            331499 non-null  object
 6   Address          380234 non-null  object
 7   Images           345209 non-null  object
 8   Categories       345153 non-null  object
 9   Geo_Coordinates  172604 non-null  object
 10  Time_Zone        172604 non-null  object
 11  Latitude         380358 non-null  object
 12  Longitude        380141 non-null  object
dtypes: object(13)
memory usage: 37.7+ MB


There are a total of 13 columns and 380,358 rows. Each row represents a restaurant listing with fields for:
- identity (Title, Category, Categories)
- quality/engagement (Rating)
- contact/presence (Website, Phone, Link, Images)
- location (Address, Latitude, Longitude, Geo_Coordinates, Time_Zone)

as columns. 

In [4]:
# Summary statistics for numeric columns
df.describe()

,Title,Link,Category,Rating,Website,Phone,Address,Images,Categories,Geo_Coordinates,Time_Zone,Latitude,Longitude
count,380358,345209,380302,337877.0,285435,331499,380234,345209,345153,172604,172604,380358.000000,380141.000000
unique,123770,156220,716,82.0,121313,149571,180187,335601,34826,154590,56,274279.000000,230191.000000
top,Subway,https://www.google.com/maps/place/Hawkers+Asia...,Restaurant,4.4,https://www.dominos.com/en/?utm_source=google&...,+1 800-786-1000,"Hawkers Asian Street Food, 1001 Park St, Jacks...",[],['Restaurant'],"{'latitude': 33.775089, 'longitude': -84.2805889}",America/New_York,30.326381,-129.942709
freq,4188,14,56966,37896.0,1124,172,14,1154,29194,12,61506,15.000000,21.000000


**Your data profile notes:**  
- The dataset has 380,358 rows and 13 columns
- Each row is a different restaurant listing with fields for identity (Title, Category, Categories), quality (rating), contact and presence (website, phone, link, images), and location (address, latitutde, longitude, geo coordinates, time zone)
- Out of the columns, Category is one that we will be focusing on. This category has 716 unique labels, from specific cruisines to broad labels
- Not all of the data in every column is complete - about 54.6% are missing in geo_coordinates and time_zone, 25% don't have websites, 12.8% don't have phone numbers, and 11.2% don't have ratings
- Ratings is originally loaded as text that is then converted to numeric. The mean rating is about 4.27, while the median is 4.3
- There are only 56 rows missing categories. However, 86,716 restaurants have a very broad category (restaurant), which hides cuisine details
- There may be some repeat entries, since duplicate Title + Address combinations are high (52.5%)
- About 61,412 rows match Asian-related category keywords
- Largest groups include Chinese, Thai, Japanese, Indian, Vietnamese



Data Cleaning and grouping

In [5]:
# check duplicate entries by combo of Title and Address
dup_rows = df[["Title", "Address"]].fillna("").duplicated().sum()
dup_pct = dup_rows / len(df) * 100

print("Total rows:", len(df))
print("Duplicate Title+Address rows:", dup_rows)
print(f"Percent duplicate: {dup_pct:.2f}%")

Total rows: 380358
Duplicate Title+Address rows: 199686
Percent duplicate: 52.50%


In [6]:
#visualize combos of Title and Address + count
(
    df.assign(Title=df["Title"].fillna(""), Address=df["Address"].fillna(""))
      .groupby(["Title", "Address"], dropna=False)
      .size()
      .reset_index(name="count")
      .sort_values("count", ascending=False)
      .head(10)
)

,Title,Address,count
64451,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",14
153372,Tasty Pot,"Tasty Pot, 9802 Baymeadows Rd, Jacksonville, F...",12
105642,Newport Tokyo House,"Newport Tokyo House, 6 Equality Park Pl, Newpo...",12
16206,Blue Orchid Thai Cuisine,"Blue Orchid Thai Cuisine, 13475 Atlantic Blvd ...",12
56895,Gangnam Korean Restaurant,"Gangnam Korean Restaurant, 5161 Beach Blvd #5,...",12
109483,Olive Garden Italian Restaurant,"Olive Garden Italian Restaurant, 4860 N Hamilt...",10
106894,Noodles and Company,"Noodles and Company, 2560 E 3rd St, Bloomingto...",10
44783,Drift Cafe,"Drift Cafe, 190 Thames St, Newport, RI 02840, ...",10
133959,Salt Life Food Shack,"Salt Life Food Shack, 1018 3rd St N, Jacksonvi...",10
71302,Izakaya Japanese Restaurant & Sake Bar,"Izakaya Japanese Restaurant & Sake Bar, 39 Col...",10


In [7]:
# 1) Build a clean duplicate key
work = df.copy()
for c in ["Title", "Address", "Phone", "Latitude", "Longitude", "Geo_Coordinates"]:
    if c in work.columns:
        work[c] = work[c].fillna("").astype(str).str.strip()

# Keep only rows where Title+Address appears more than once
dup_mask = work.duplicated(subset=["Title", "Address"], keep=False)
dups = work.loc[dup_mask].copy()

print("Duplicate rows (Title+Address):", len(dups))
print("Unique duplicate groups:", dups.groupby(["Title", "Address"]).ngroups)

Duplicate rows (Title+Address): 367492
Unique duplicate groups: 167806


In [8]:
# 2) For each duplicate group, count unique values in each field
group_check = (
    dups.groupby(["Title", "Address"], dropna=False)
        .agg(
            rows=("Title", "size"),
            phone_unique=("Phone", "nunique"),
            lat_unique=("Latitude", "nunique"),
            lon_unique=("Longitude", "nunique"),
            geo_unique=("Geo_Coordinates", "nunique"),
        )
        .reset_index()
)

# Flags: True means values differ within the same Title+Address group
group_check["phone_mismatch"] = group_check["phone_unique"] > 1
group_check["lat_mismatch"] = group_check["lat_unique"] > 1
group_check["lon_mismatch"] = group_check["lon_unique"] > 1
group_check["geo_mismatch"] = group_check["geo_unique"] > 1

# Summary
summary = {
    "groups_total": len(group_check),
    "phone_mismatch_groups": int(group_check["phone_mismatch"].sum()),
    "lat_mismatch_groups": int(group_check["lat_mismatch"].sum()),
    "lon_mismatch_groups": int(group_check["lon_mismatch"].sum()),
    "geo_mismatch_groups": int(group_check["geo_mismatch"].sum()),
}
summary

{'groups_total': 167806,
 'phone_mismatch_groups': 2192,
 'lat_mismatch_groups': 218,
 'lon_mismatch_groups': 219,
 'geo_mismatch_groups': 149807}

In [9]:
# 3) Show examples where something differs
examples = group_check[
    group_check[["phone_mismatch", "lat_mismatch", "lon_mismatch", "geo_mismatch"]].any(axis=1)
].sort_values("rows", ascending=False).head(10)

examples

,Title,Address,rows,phone_unique,lat_unique,lon_unique,geo_unique,phone_mismatch,lat_mismatch,lon_mismatch,geo_mismatch
60118,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",14,1,1,1,2,False,False,False,True
98239,Newport Tokyo House,"Newport Tokyo House, 6 Equality Park Pl, Newpo...",12,1,1,1,2,False,False,False,True
142474,Tasty Pot,"Tasty Pot, 9802 Baymeadows Rd, Jacksonville, F...",12,1,1,1,2,False,False,False,True
15094,Blue Orchid Thai Cuisine,"Blue Orchid Thai Cuisine, 13475 Atlantic Blvd ...",12,1,1,1,2,False,False,False,True
52975,Gangnam Korean Restaurant,"Gangnam Korean Restaurant, 5161 Beach Blvd #5,...",12,1,1,1,2,False,False,False,True
45939,Eleven South,"Eleven South, 216 11th Ave S, Jacksonville Bea...",10,1,1,1,2,False,False,False,True
57343,Grill House,"Grill House, Food Court, 21182 Salmon Run Mall...",10,1,1,1,2,False,False,False,True
83784,Lucky Star Restaurant,"Lucky Star Restaurant, 1140 Meadow Ln, Concord...",10,1,1,1,2,False,False,False,True
101820,Olive Garden Italian Restaurant,"Olive Garden Italian Restaurant, 4860 N Hamilt...",10,1,1,1,2,False,False,False,True
52222,Fujiyama Japanese Restaurant,"Fujiyama Japanese Restaurant, 21 Regent St, Cl...",10,1,1,1,2,False,False,False,True


In [12]:
# 4) Drill into one mismatched group to inspect exact values
# (run after viewing examples)
from IPython.display import display

print("examples rows:", len(examples))

if len(examples) == 0:
    print("No mismatched duplicate groups found.")
else:
    t = examples.iloc[0]["Title"]
    a = examples.iloc[0]["Address"]
    print("Inspecting group:")
    print("Title:", t)
    print("Address:", a)

    out = dups[(dups["Title"] == t) & (dups["Address"] == a)][
        ["Title", "Address", "Phone", "Latitude", "Longitude", "Geo_Coordinates", "Link", "Categories"]
    ].head(20)

    print("Rows shown:", len(out))
    display(out)

examples rows: 10
Inspecting group:
Title: Hawkers Asian Street Food
Address: Hawkers Asian Street Food, 1001 Park St, Jacksonville, FL 32204, United States
Rows shown: 14


,Title,Address,Phone,Latitude,Longitude,Geo_Coordinates,Link,Categories
7171,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,"{'latitude': 30.315006, 'longitude': -81.680353}",https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
34533,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,"{'latitude': 30.315006, 'longitude': -81.680353}",https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
45401,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,"{'latitude': 30.315006, 'longitude': -81.680353}",https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
109804,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,"{'latitude': 30.315006, 'longitude': -81.680353}",https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
116979,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,"{'latitude': 30.315006, 'longitude': -81.680353}",https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
160200,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,"{'latitude': 30.315006, 'longitude': -81.680353}",https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
183600,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,"{'latitude': 30.315006, 'longitude': -81.680353}",https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
252230,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,,https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
314495,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,,https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
350767,Hawkers Asian Street Food,"Hawkers Asian Street Food, 1001 Park St, Jacks...",+1 904-508-0342,30.315006,-81.680353,,https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']


In [13]:
subset = dups[(dups["Title"] == t) & (dups["Address"] == a)][
    ["Phone", "Latitude", "Longitude", "Geo_Coordinates", "Link", "Categories"]
]

print("Unique Phone:", subset["Phone"].nunique())
print("Unique Lat:", subset["Latitude"].nunique())
print("Unique Lon:", subset["Longitude"].nunique())
print("Unique Geo:", subset["Geo_Coordinates"].nunique())

subset.drop_duplicates()

Unique Phone: 1
Unique Lat: 1
Unique Lon: 1
Unique Geo: 2


,Phone,Latitude,Longitude,Geo_Coordinates,Link,Categories
7171,+1 904-508-0342,30.315006,-81.680353,"{'latitude': 30.315006, 'longitude': -81.680353}",https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']
252230,+1 904-508-0342,30.315006,-81.680353,,https://www.google.com/maps/place/Hawkers+Asia...,['Asian restaurant']


After cleaning data, there are 183,727 rows left. 

In [15]:
import pandas as pd

# Load cleaned dataset
df = pd.read_csv("restaurants_cleaned.csv", low_memory=False)

# Define Asian-related keyword pattern (word-boundary to avoid matching "caucasian")
pattern = r"\b(?:asian|chinese|japanese|thai|vietnamese|korean|indian|filipino|malaysian|indonesian|taiwanese)\b"

# Build filtered dataframe
asian_df = df[
    df["Category"].fillna("").str.lower().str.contains(pattern, regex=True)
].copy()

# Explicitly remove "Caucasian restaurant" just in case
asian_df = asian_df[
    asian_df["Category"].fillna("").str.lower() != "caucasian restaurant"
]

# Summary stats
total_rows = len(df)
asian_rows = len(asian_df)
asian_pct = (asian_rows / total_rows) * 100
num_categories = asian_df["Category"].nunique(dropna=True)

print("Total rows:", total_rows)
print("Asian-related rows (excluding Caucasian restaurant):", asian_rows)
print(f"Percent Asian-related: {asian_pct:.2f}%")
print("Unique Asian-related categories:", num_categories)

# Show categories counted
categories_used = sorted(asian_df["Category"].dropna().astype(str).str.strip().unique())
print("\nCategories included:")
for c in categories_used:
    print("-", c)

# Optional: top categories by count
print("\nTop 20 Asian-related categories:")
print(asian_df["Category"].value_counts().head(20))

Total rows: 183727
Asian-related rows (excluding Caucasian restaurant): 27195
Percent Asian-related: 14.80%
Unique Asian-related categories: 39

Categories included:
- Asian fusion restaurant
- Asian grocery store
- Asian restaurant
- Authentic Japanese restaurant
- Chinese bakery
- Chinese noodle restaurant
- Chinese restaurant
- Chinese supermarket
- Chinese takeaway
- Chinese tea house
- Delivery Chinese restaurant
- Filipino restaurant
- Indian Muslim restaurant
- Indian grocery store
- Indian restaurant
- Indian takeaway
- Indonesian restaurant
- Japanese curry restaurant
- Japanese delicatessen
- Japanese grocery store
- Japanese regional restaurant
- Japanese restaurant
- Japanese steakhouse
- Korean barbecue restaurant
- Korean grocery store
- Korean restaurant
- Korean rib restaurant
- Kyoto style Japanese restaurant
- Malaysian restaurant
- Modern Indian restaurant
- North Eastern Indian restaurant
- North Indian restaurant
- Pan-Asian restaurant
- South Asian restaurant
- So

In the cleaned dataset, there are 27,196 rows, with 40 unique categories that are Asian related. 

---

## Section 3 — Analysis

Answer your three research questions using pandas. Each question should have:

1. A markdown cell stating the question
2. A code cell with the analysis
3. A markdown cell with your interpretation — what does the result mean?

You may need to clean or reshape the data before you can answer a question. That's normal — document what you did and why.

**Question 1:** Which Asian cuisines (e.g., Chinese, Japanese, Thai, Vietnamese, etc.) are most and least represented among restaurants in the dataset?


In [16]:
# Your analysis for Question 1
# Question 1:
# Which Asian cuisines are most and least represented?

import pandas as pd

# Load cleaned file (created earlier)
df = pd.read_csv("restaurants_cleaned.csv", low_memory=False)

# 1) Keep only rows with a category
work = df.copy()
work["Category"] = work["Category"].fillna("").astype(str).str.strip()
work = work[work["Category"] != ""]

# 2) Map raw category labels -> cuisine groups
# (edit/expand this mapping if you want to include more labels)
cuisine_map = {
    "Chinese": [
        "Chinese restaurant", "Chinese takeaway", "Chinese noodle restaurant",
        "Chinese bakery", "Chinese tea house", "Delivery Chinese restaurant"
    ],
    "Japanese": [
        "Japanese restaurant", "Authentic Japanese restaurant",
        "Japanese steakhouse", "Japanese curry restaurant",
        "Japanese regional restaurant", "Japanese delicatessen",
        "Kyoto style Japanese restaurant"
    ],
    "Thai": ["Thai restaurant"],
    "Vietnamese": ["Vietnamese restaurant"],
    "Korean": [
        "Korean restaurant", "Korean barbecue restaurant", "Korean rib restaurant"
    ],
    "Indian": [
        "Indian restaurant", "North Indian restaurant", "South Indian restaurant",
        "North Eastern Indian restaurant", "Modern Indian restaurant",
        "Indian Muslim restaurant", "Indian takeaway"
    ],
    "Filipino": ["Filipino restaurant"],
    "Malaysian": ["Malaysian restaurant"],
    "Indonesian": ["Indonesian restaurant"],
    "Taiwanese": ["Taiwanese restaurant"],
    # Optional broader buckets:
    "Pan-Asian / General Asian": [
        "Asian restaurant", "Asian fusion restaurant",
        "Pan-Asian restaurant", "South Asian restaurant", "Southeast Asian restaurant"
    ],
}

# Build reverse lookup (raw category -> cuisine group)
reverse_map = {}
for cuisine, labels in cuisine_map.items():
    for label in labels:
        reverse_map[label.lower()] = cuisine

# 3) Assign cuisine group
work["Cuisine_Group"] = work["Category"].str.lower().map(reverse_map)

# Keep only mapped Asian cuisines
asian = work[work["Cuisine_Group"].notna()].copy()

# 4) Count representation
counts = (
    asian["Cuisine_Group"]
    .value_counts()
    .rename_axis("Cuisine")
    .reset_index(name="Restaurant_Count")
)

# Add percent share within Asian subset
counts["Percent_of_Asian"] = (counts["Restaurant_Count"] / counts["Restaurant_Count"].sum() * 100).round(2)

# Most represented
print("Most represented cuisines:")
display(counts.head(5))

# Least represented (non-zero)
print("Least represented cuisines:")
display(counts.tail(5))

# Full table
display(counts)

Most represented cuisines:


,Cuisine,Restaurant_Count,Percent_of_Asian
0,Chinese,10845,40.11
1,Japanese,3631,13.43
2,Thai,3557,13.15
3,Pan-Asian / General Asian,3270,12.09
4,Indian,2212,8.18


Least represented cuisines:


,Cuisine,Restaurant_Count,Percent_of_Asian
6,Korean,1097,4.06
7,Filipino,268,0.99
8,Taiwanese,94,0.35
9,Malaysian,45,0.17
10,Indonesian,34,0.13


,Cuisine,Restaurant_Count,Percent_of_Asian
0,Chinese,10845,40.11
1,Japanese,3631,13.43
2,Thai,3557,13.15
3,Pan-Asian / General Asian,3270,12.09
4,Indian,2212,8.18
5,Vietnamese,1988,7.35
6,Korean,1097,4.06
7,Filipino,268,0.99
8,Taiwanese,94,0.35
9,Malaysian,45,0.17


**Interpretation:**  
Asian cuisine representation in this dataset is uneven. Chinese restaurants are the most represented category by a wide margin, followed by Thai and Japanese, while cuisines such as Indonesian, Malaysian, and Filipino appear much less frequently. This suggests that some Asian cuisines are much more visible on this platform than others, which could reflect differences in real-world restaurant prevalence, platform coverage, or category labeling practices. A key limitation is that cuisine labels are not perfectly standardized (for example, broad labels like “Asian restaurant” and mixed labels like “Asian fusion”), so these counts should be interpreted as approximate representation rather than exact cultural market share.


**Question 2:** How does the representation of Asian cuisines vary across different U.S. states (e.g., which states have higher or lower proportions of Asian cuisine restaurants)?


In [17]:
# Your analysis for Question 2
# Question 2:
# How does representation of Asian cuisines vary across U.S. states?

import pandas as pd

# Load cleaned dataset
df = pd.read_csv("restaurants_cleaned.csv", low_memory=False)

# -----------------------------
# 1) Create an Asian flag
# -----------------------------
pattern = r"\b(?:asian|chinese|japanese|thai|vietnamese|korean|indian|filipino|malaysian|indonesian|taiwanese)\b"

work = df.copy()
work["Category"] = work["Category"].fillna("").astype(str).str.strip()
work["is_asian"] = work["Category"].str.lower().str.contains(pattern, regex=True)

# Exclude false positive explicitly
work.loc[work["Category"].str.lower() == "caucasian restaurant", "is_asian"] = False

# -----------------------------
# 2) Extract U.S. state from Address
# -----------------------------
# Example format: "... , Seattle, WA 98101, United States"
work["Address"] = work["Address"].fillna("").astype(str)
work["State"] = work["Address"].str.extract(
    r",\s*([A-Z]{2})\s+\d{5}(?:-\d{4})?,\s*United States\s*$"
)[0]

# Keep only rows where state was extracted
state_df = work[work["State"].notna()].copy()

# -----------------------------
# 3) State-level representation
# -----------------------------
state_summary = (
    state_df.groupby("State", as_index=False)
    .agg(
        total_restaurants=("State", "size"),
        asian_restaurants=("is_asian", "sum")
    )
)

state_summary["asian_share_pct"] = (
    state_summary["asian_restaurants"] / state_summary["total_restaurants"] * 100
).round(2)

# Optional: require minimum sample size for fair comparison
state_summary_filtered = state_summary[state_summary["total_restaurants"] >= 500].copy()

# Highest and lowest shares
highest = state_summary_filtered.sort_values("asian_share_pct", ascending=False).head(10)
lowest = state_summary_filtered.sort_values("asian_share_pct", ascending=True).head(10)

print("Top 10 states by Asian restaurant share (min 500 total restaurants):")
display(highest)

print("Bottom 10 states by Asian restaurant share (min 500 total restaurants):")
display(lowest)

Top 10 states by Asian restaurant share (min 500 total restaurants):


,State,total_restaurants,asian_restaurants,asian_share_pct
4,CA,17405,5007,28.77
11,HI,1303,335,25.71
3,AZ,2479,634,25.57
10,GA,2433,609,25.03
9,FL,7736,1766,22.83
0,AK,807,165,20.45
47,WA,3330,630,18.92
5,CO,3727,690,18.51
6,CT,3713,631,16.99
14,IL,6412,980,15.28


Bottom 10 states by Asian restaurant share (min 500 total restaurants):


,State,total_restaurants,asian_restaurants,asian_share_pct
41,SD,689,42,6.10
28,ND,853,52,6.10
50,WY,808,55,6.81
32,NM,1452,103,7.09
49,WV,1449,107,7.38
36,OK,2739,203,7.41
17,KY,1383,105,7.59
26,MT,943,73,7.74
25,MS,1549,120,7.75
23,MN,1637,129,7.88


**Interpretation:**  
Representation of Asian cuisines varies substantially across states. Some states have a much higher proportion of Asian cuisine restaurants, while others have relatively low shares, indicating uneven geographic visibility and concentration. This suggests that local demographics, migration patterns, and regional food markets likely influence cuisine representation. A limitation is that state extraction depends on address formatting and category labels, so results are best interpreted as comparative patterns rather than exact market estimates.

**Question 3:** Do average ratings differ across Asian cuisine types, and are certain cuisines consistently rated higher or lower than others?

In [18]:
# Your analysis for Question 3
# Question 3:
# Do average ratings differ across Asian cuisine types?

import pandas as pd

# Load cleaned dataset
df = pd.read_csv("restaurants_cleaned.csv", low_memory=False)

# Keep needed columns and clean
work = df[["Category", "Rating"]].copy()
work["Category"] = work["Category"].fillna("").astype(str).str.strip()
work["rating_num"] = pd.to_numeric(work["Rating"], errors="coerce")

# Keep only valid ratings
work = work[work["rating_num"].notna()]

# Map raw category labels -> cuisine groups
cuisine_map = {
    "Chinese": [
        "Chinese restaurant", "Chinese takeaway", "Chinese noodle restaurant",
        "Chinese bakery", "Chinese tea house", "Delivery Chinese restaurant"
    ],
    "Japanese": [
        "Japanese restaurant", "Authentic Japanese restaurant",
        "Japanese steakhouse", "Japanese curry restaurant",
        "Japanese regional restaurant", "Japanese delicatessen",
        "Kyoto style Japanese restaurant"
    ],
    "Thai": ["Thai restaurant"],
    "Vietnamese": ["Vietnamese restaurant"],
    "Korean": [
        "Korean restaurant", "Korean barbecue restaurant", "Korean rib restaurant"
    ],
    "Indian": [
        "Indian restaurant", "North Indian restaurant", "South Indian restaurant",
        "North Eastern Indian restaurant", "Modern Indian restaurant",
        "Indian Muslim restaurant", "Indian takeaway"
    ],
    "Filipino": ["Filipino restaurant"],
    "Malaysian": ["Malaysian restaurant"],
    "Indonesian": ["Indonesian restaurant"],
    "Taiwanese": ["Taiwanese restaurant"],
    "Pan-Asian / General Asian": [
        "Asian restaurant", "Asian fusion restaurant",
        "Pan-Asian restaurant", "South Asian restaurant", "Southeast Asian restaurant"
    ],
}

# Reverse mapping
reverse_map = {}
for cuisine, labels in cuisine_map.items():
    for label in labels:
        reverse_map[label.lower()] = cuisine

work["Cuisine_Group"] = work["Category"].str.lower().map(reverse_map)

# Keep mapped cuisines only
asian_ratings = work[work["Cuisine_Group"].notna()].copy()

# Aggregate stats by cuisine
q3_summary = (
    asian_ratings.groupby("Cuisine_Group", as_index=False)
    .agg(
        n_restaurants=("rating_num", "size"),
        avg_rating=("rating_num", "mean"),
        median_rating=("rating_num", "median"),
        std_rating=("rating_num", "std")
    )
)

# Keep cuisines with enough observations for stability
q3_summary = q3_summary[q3_summary["n_restaurants"] >= 50].copy()

# Sort by average rating
q3_summary = q3_summary.sort_values("avg_rating", ascending=False)
q3_summary["avg_rating"] = q3_summary["avg_rating"].round(3)
q3_summary["median_rating"] = q3_summary["median_rating"].round(3)
q3_summary["std_rating"] = q3_summary["std_rating"].round(3)

display(q3_summary)

print("Highest average-rated cuisines:")
display(q3_summary.head(5))

print("Lowest average-rated cuisines:")
display(q3_summary.tail(5))

,Cuisine_Group,n_restaurants,avg_rating,median_rating,std_rating
9,Thai,3534,4.396,4.4,0.287
10,Vietnamese,1979,4.375,4.4,0.253
5,Korean,1093,4.365,4.4,0.310
4,Japanese,3582,4.358,4.4,0.305
1,Filipino,234,4.327,4.4,0.508
8,Taiwanese,92,4.312,4.3,0.256
2,Indian,2181,4.294,4.3,0.342
7,Pan-Asian / General Asian,3125,4.272,4.3,0.453
0,Chinese,10721,4.048,4.1,0.388


Highest average-rated cuisines:


,Cuisine_Group,n_restaurants,avg_rating,median_rating,std_rating
9,Thai,3534,4.396,4.4,0.287
10,Vietnamese,1979,4.375,4.4,0.253
5,Korean,1093,4.365,4.4,0.310
4,Japanese,3582,4.358,4.4,0.305
1,Filipino,234,4.327,4.4,0.508


Lowest average-rated cuisines:


,Cuisine_Group,n_restaurants,avg_rating,median_rating,std_rating
1,Filipino,234,4.327,4.4,0.508
8,Taiwanese,92,4.312,4.3,0.256
2,Indian,2181,4.294,4.3,0.342
7,Pan-Asian / General Asian,3125,4.272,4.3,0.453
0,Chinese,10721,4.048,4.1,0.388


**Interpretation:**  
Average ratings do differ across Asian cuisine types, but the differences are relatively modest on a 1–5 scale. Some cuisine groups consistently appear near the top of average ratings, while others are lower, suggesting variation in customer satisfaction (or reviewing behavior) across cuisine categories. To avoid over-interpreting small groups, I applied a minimum sample-size threshold (n >= 50) before comparing means. A limitation is that ratings are platform-generated and may be influenced by location, chain effects, and review volume, so these results show comparative patterns rather than causal quality differences.

---

## Section 4 — Visualization

Create at least one visualization that supports one of your analysis findings. Your chart should:

- Have a title that states the finding, not just the data (e.g., "Satisfaction scores drop sharply after age 40" not "Satisfaction by age")
- Have labeled axes
- Use a chart type appropriate for your data (bar for categorical comparison, scatter for relationships, line for trends over time)

Below the chart, explain in a markdown cell: why you chose this chart type, and what you want the reader to take away from it.

Which Asian cuisines (e.g., Chinese, Japanese, Thai, Vietnamese, etc.) are most and least represented among restaurants in the dataset?

Chart for Q1

In [22]:
# Your visualization
# Example structure — replace with your actual columns and finding

# fig = px.bar(
#     df,
#     x="your_category_column",
#     y="your_value_column",
#     title="Your finding stated as a claim",
#     labels={"your_category_column": "Readable label", "your_value_column": "Readable label"}
# )
# fig.show()

#most + least represented chart
import textwrap

top5 = counts.nlargest(5, "Restaurant_Count")
bottom5 = counts.nsmallest(5, "Restaurant_Count")

top_bottom = (
    pd.concat([top5, bottom5], ignore_index=True)
      .drop_duplicates(subset="Cuisine")
      .sort_values("Restaurant_Count")
)

_title = (
    "Chinese, Japanese, Thai, general Asian, Indian, Korean, and Filipino categories "
    "dominate category counts"
)
_title_wrapped = "<br>".join(textwrap.wrap(_title, width=62))

fig = px.bar(
    top_bottom,
    x="Restaurant_Count",
    y="Cuisine",
    orientation="h",
    color="Cuisine",
    title=_title_wrapped,
    labels={"Restaurant_Count": "Number of restaurants", "Cuisine": "Cuisine type"},
)
fig.update_layout(margin=dict(t=100))
fig.show()
fig.write_image("most_and_least_represented_asian_cuisines.png")

**Chart rationale:**  
I used a “top 5 vs. bottom 5” horizontal bar chart to directly compare the most and least represented Asian cuisine types in the dataset. This format makes the contrast clear without overwhelming the reader with every category. The chart shows that cuisine representation is highly uneven: a few cuisines account for much larger shares, while several cuisines appear relatively rarely. This supports the conclusion that visibility of Asian cuisines on the platform is not evenly distributed.



How does the representation of Asian cuisines vary across different U.S. states (e.g., which states have higher or lower proportions of Asian cuisine restaurants)?

Chart for Q2

In [26]:
#cholopleth of asian restaurants by state
import pandas as pd
import plotly.express as px

# --- Load data ---
df = pd.read_csv("restaurants_cleaned.csv", low_memory=False)

# --- Asian flag (word-boundary; excludes Caucasian false positive) ---
pattern = r"\b(?:asian|chinese|japanese|thai|vietnamese|korean|indian|filipino|malaysian|indonesian|taiwanese)\b"
work = df.copy()
work["Category"] = work["Category"].fillna("").astype(str).str.strip()
work["is_asian"] = work["Category"].str.lower().str.contains(pattern, regex=True)
work.loc[work["Category"].str.lower() == "caucasian restaurant", "is_asian"] = False

# --- US state from address: ", ST 12345, United States" ---
work["Address"] = work["Address"].fillna("").astype(str)
work["State"] = work["Address"].str.extract(
    r",\s*([A-Z]{2})\s+\d{5}(?:-\d{4})?,\s*United States\s*$"
)[0]

us = work[work["State"].notna()].copy()

# --- Aggregate: Asian restaurant count per state ---
choropleth = (
    us.groupby("State", as_index=False)
      .agg(
          total_restaurants=("State", "size"),
          asian_restaurants=("is_asian", "sum"),
      )
)

choropleth["asian_share_pct"] = (
    choropleth["asian_restaurants"] / choropleth["total_restaurants"] * 100
).round(2)

# --- Choropleth: COUNT of Asian restaurants ---
fig = px.choropleth(
    choropleth,
    locations="State",
    locationmode="USA-states",
    color="asian_restaurants",
    scope="usa",
    title="California has the most number of Asian cuisine restaurants purely by count",
    labels={"asian_restaurants": "Asian restaurants"},
    color_continuous_scale="Blues",
)
fig.update_layout(
    margin=dict(l=0, r=0, t=50, b=0),
    geo=dict(scope="usa", projection_scale=1.05),
)
fig.show()
fig.write_image("asian_cuisine_restaurants_by_state.png")

In [25]:
#cholopleth by percent
fig2 = px.choropleth(
    choropleth,
    locations="State",
    locationmode="USA-states",
    color="asian_share_pct",
    scope="usa",
    title="Asian cuisine’s share of each state’s listings tells a different story than raw counts alone",
    labels={"asian_share_pct": "Asian share (%)"},
    color_continuous_scale="Reds",
)
fig2.update_layout(margin=dict(l=0, r=0, t=50, b=0), geo=dict(scope="usa"))
fig2.show()
fig2.write_image("share_of_restaurants_that_are_asian_cuisine.png")

**Chart rationale:**  
I used two choropleth maps side by side because they answer different but related questions about the same geography. The first map plots the number of Asian-cuisine restaurants per state; it highlights where the largest totals appear (often states with bigger populations and more listings overall). The second map plots share (% of restaurants in each state flagged as Asian cuisine); it adjusts for scale and makes it easier to compare whether Asian cuisine listings are unusually common or uncommon in smaller states versus larger ones.

Together, these views reduce misinterpretation: a state can rank high on counts but only moderate on share, meaning “many listings” partly reflects marketplace size—not only cultural food presence. Readers should interpret both maps relative to limitations in the dataset (category labels as proxies for cuisine and address-based state parsing dropping some rows). The takeaway should combine both signals: geographic concentration and proportional representation within each state’s slice of the data.

Do average ratings differ across Asian cuisine types, and are certain cuisines consistently rated higher or lower than others?

Chart for Q3


In [30]:
import plotly.express as px

fig = px.bar(
    q3_summary.sort_values("avg_rating", ascending=False),
    x="Cuisine_Group",
    y="avg_rating",
    color="n_restaurants",
    title="Mean ratings by cuisine are very similar",
    labels={"avg_rating": "Average rating", "Cuisine_Group": "Cuisine type", "n_restaurants": "Sample size"}
)
fig.update_layout(xaxis_tickangle=-35)
fig.show()

In [31]:
import numpy as np
import pandas as pd
import plotly.express as px

rng = np.random.default_rng(42)
# Per-restaurant rows with mapped cuisine (defined in the Question 3 prep cell)
plot = asian_ratings.copy()
cat = pd.Categorical(plot["Cuisine_Group"])
cuisine_order = list(cat.categories)
plot["xj"] = cat.codes.astype(float) + rng.uniform(-0.25, 0.25, size=len(plot))

fig = px.scatter(
    plot,
    x="xj",
    y="rating_num",
    color="Cuisine_Group",
    labels={"rating_num": "Rating (1–5)", "xj": "Cuisine type (jittered)"},
    title="Restaurant-level ratings overlap heavily across cuisines despite small differences in averages",
)

fig.update_traces(marker=dict(size=6, opacity=0.25))
fig.update_layout(
    height=700,
    showlegend=True,
    xaxis=dict(
        title="Cuisine type (horizontal jitter for visibility)",
        tickmode="array",
        tickvals=list(range(len(cuisine_order))),
        ticktext=cuisine_order,
        tickangle=-35,
    ),
)
fig.show()

NameError: name 'strip_df' is not defined

**Chart Rationale**
I used a jittered strip scatter plot instead of only plotting averages because ratings rarely behave like a single number—within each cuisine type many restaurants cluster near typical ratings while others fall higher or lower. Spreading points horizontally with jitter separates overlapping markers so density is visible rather than mashed into one vertical stack. Transparency keeps large groups readable without implying false precision about individual listings. Together, this chart supports the Question 3 claim that ratings differ between cuisines and clarifies whether differences look like a consistent shift upward/downward versus wide overlap in the distributions.



---

## Section 5 — Conclusions

Write 3–5 sentences summarizing what you found. Address these questions:

- What is the most important thing your analysis revealed?
- What surprised you?
- What would you investigate next if you had more time or data?
- What are the limitations of this analysis — what can't you conclude from this data?

Then complete the competency claim below.


This analysis used a deduplicated restaurant dataset to compare how Asian cuisines appear in platform category labels across the United States and how ratings differ by mapped cuisine groups. Cuisine representation is uneven: Chinese, Thai, and Japanese categories appear among the largest groups, while several other Asian cuisines occur much less often. Geography matters in two ways: some states show many Asian-cuisine listings in absolute terms, but state-level shares (percent of listings) tell a different story about proportional visibility. Ratings differ modestly across cuisine groups, though distributions overlap—so averages should be read cautiously alongside spread and sample size.

- **What is the most important thing your analysis revealed?**
The dataset does not reflect Asian cuisines evenly: a few cuisines dominate category counts while others are relatively rare. That imbalance affects how visible different cultures appear on this kind of listings platform—not only where restaurants exist.

- **What surprised you?**
(Pick what matched your plots.) I was struck by how large the gap can be between the most and least represented cuisines after cleaning duplicates, and by how much counts vs. state share can disagree when comparing states.

- **What would you investigate next if you had more time or data?**
I would add review count and price level (if available), compare metro areas instead of only states, and improve location parsing with coordinates. I would also test a stricter cuisine taxonomy (e.g., using Categories or external labels) to reduce reliance on a single Category string.

- **What are the limitations — what can't you conclude?**
You cannot claim true population demographics or causal “food quality” from star ratings alone. Categories are platform-defined and inconsistent; Pan-Asian / Restaurant buckets blur cuisine identity; Address-based states omit some rows; deduping rules (Title+Address+Phone+Lat+Lon) change totals; ratings can reflect chains, reviewer bias, listing completeness, and missing ratings. At best this analysis shows patterns in scraped/listing labels and aggregates, not a complete census of U.S. dining.

**Summary of findings:**  
*(Write your 3–5 sentence conclusion here.)*

---

## Competency Claim

In a `mp1.md` file in your GitHub repository, write a short competency claim (2–4 sentences) for each domain you feel this project demonstrates. Be specific — cite something you actually did in this notebook.

Domains covered by this project typically include:
- **C3 — Data cleaning and file handling** (if you cleaned or reshaped data)
- **C5 — Data analysis with pandas** (answering questions with code)
- **C6 — Data visualization** (your chart)
- **C7 — Critical evaluation and professional judgment** (your interpretation and limitations section)

You don't have to claim every domain — only the ones your work actually demonstrates.